lib/pos_mempool.go

```diff
@@ -244,6 +251,18 @@ func (mp *PosMempool) Init(
 	mp.mempoolBackupIntervalMillis = mempoolBackupIntervalMillis
 	mp.maxValidationViewConnects = maxValidationViewConnects
 	mp.transactionValidationRefreshIntervalMillis = transactionValidationRefreshIntervalMillis
+	mp.recentBlockTxnCache = lru.NewKVCache(10000) // TODO: parameterize?
+
+	// Add all the transactions in the past blocks to the recentBlockTxnCache.
+	for _, block := range feeEstimatorPastBlocks {
+		for _, txn := range block.Txns {
+			txnHash := txn.Hash()
+			if txnHash == nil {
+				continue
+			}
+			mp.addTxnHashToRecentBlockCache(*txnHash)
+		}
+	}
```

Comment:
Not necessary to initialize the cache on construction. I'd even opt to remove this. This is only called when the node first boots up.

The cases where we want txn hashes in the caches are when a user submits a txn to our mempool and we want to make sure the txn ended up in a block eventually.